File: designpatternsCpp/src/behavior/memento/Caretaker.cpp
/*
 * Caretaker.cpp
 *
 *  Created on: 2015Äê4ÔÂ7ÈÕ
 *      Author: renho
 */

#include "Caretaker.h"

Caretaker::Caretaker() {
}

Caretaker::~Caretaker() {
}

void Caretaker::setMemento(Memento* m) {
	memento = m;
}

Memento* Caretaker::getMemento() {
	return memento;
}

File: designpatterns/src/com/renho/behavior/memento/ex/Caretaker.java
package com.renho.behavior.memento.ex;

public class Caretaker {
	private MementoIF memento;

	public MementoIF retrieveMemento() {
		return this.memento;
	}

	public void saveMemento(MementoIF memento) {
		this.memento = memento;
	}
}
